src/coreclr/src/vm/typehandle.inl

```diff
@@ -38,48 +38,33 @@ inline void TypeHandle::SetIsFullyLoaded()
         return AsMethodTable()->SetIsFullyLoaded();
 }
 
-inline MethodTable* TypeHandle::GetMethodTableOfElementType() const
+inline MethodTable* TypeHandle::GetMethodTableOfRootElementType() const
 {
     LIMITED_METHOD_CONTRACT;
 
-    if (IsTypeDesc())
-    {
-        TypeHandle elementType = AsTypeDesc()->GetTypeParam();
-        return elementType.GetMethodTableOfElementType();
-    }
-    else
-    {
-        return AsMethodTable();
-    }
+    TypeHandle current = *this;
+    while (current.HasTypeParam())
```

Comment:
What about array of pointers? This method will return the type that the pointer points to - not an element type.

Reply:
I think this is intentional. The helper is used for accessibility checks and such. It basically digs through constructed types to the concrete type parameter to delegate the checks. 
It should unwrap array of pointers to the pointed type.

Reply:
Yes, I agree. The method should be called ...RootTypeParam... then